api/src/main/java/jakarta/servlet/http/HttpServlet.java

```diff
@@ -586,6 +590,9 @@ public void service(ServletRequest req, ServletResponse res) throws ServletExcep
 
         service(request, response);
     }
+
+    public interface HeadHandledByContainer extends ServletConfig {
```

Comment:
I don't really like this, IMHO we should deprecate the NoBody stuff and stipulate that it is the containers responsibility (although I may be slightly biased as I know this will 'just work' in Undertow, not sure about other containers).

If we did this would could have a jakarta.servlet.legacy-head-handling as a SC param to re-enable the legacy behavior.

Reply:
Ok, I'll give that a try...